### PR TITLE
Add Async::new_nonblocking and avoid needless set_nonblocking calls

### PR DESCRIPTION
**Pull request by taiki-e**
connect() already sets socket to non-blocking mode.

The second comment exposes Async::new_nonblocking to resolve https://github.com/smol-rs/async-io/issues/132.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -731,16 +733,18 @@ impl<T: AsSocket> Async<T> {
...
-        let borrowed = io.as_socket();
-
         // Put the socket in non-blocking mode.
-        set_nonblocking(borrowed)?;
+        set_nonblocking(io.as_socket())?;
+
+        Self::new_nonblocking(io)
+    }
 
+    fn new_nonblocking(io: T) -> io::Result<Async<T>> {
```
Comment (notgull) [inline src/lib.rs:742]: Can we make this public to solve https://github.com/smol-rs/async-io/issues/132?

**Review by notgull — APPROVED** [review]
Thanks!

File: src/lib.rs (hunk 1 above)
Comment (taiki-e) [reply src/lib.rs]: Done in https://github.com/smol-rs/async-io/pull/159/commits/de07b6adca6432e2fde5ce2a15c66c41731d7992.

File: src/lib.rs [hunk 2]
```diff
@@ -731,16 +750,35 @@ impl<T: AsSocket> Async<T> {
...
+
+        Self::new_nonblocking(io)
+    }
 
+    /// Creates an async I/O handle without setting it to non-blocking mode.
+    ///
+    /// This method will register the handle in [epoll]/[kqueue]/[event ports]/[IOCP].
+    ///
+    /// On Unix systems, the handle must implement `AsRawFd`, while on Windows it must implement
+    /// `AsRawSocket`.
```
Comment (notgull) [inline src/lib.rs:764]: Nit: This is supposed to be `AsFd` and `AsSocket`.

Comment (taiki-e) [reply src/lib.rs]: Good catch. Fixed.

**Review by notgull — APPROVED** [review]
Thanks!

**Pull request merged**